### PR TITLE
fix: address a number of typos and grammar issues

### PR DESCRIPTION
**Pull request by G-Rath**
### Summary

This addresses a number of typos and grammar issues.

### Pull Request checklist
_Remove this line after checking all the items here. If the item is not applicable to the PR, both check it out and wrap it by `~`._

- [x] ~Add/update test to cover these changes~
- [x] ~Update documentation~
- [x] Update CHANGELOG file  
  _Add the CHANGELOG entry at the top of the file._

### Other Information

I did find `ReactOnRails::WebpackerUtils.shackapacker_version_requirement_met?` but I've not corrected that as I think technically it's a public function

- - -
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1631)

## Summary by CodeRabbit

- **Documentation**
  - Fixed numerous typos and grammatical errors across multiple documentation files for improved clarity and accuracy.
  - Updated explanations and corrected configuration settings in various guides to ensure accurate functionality descriptions.

- **Bug Fixes**
  - Corrected asset symlinking to handle filenames with spaces properly.
  - Resolved ID collisions in DOM nodes generated by `react_component` by switching to UUIDs.

- **Chores**
  - Improved internal comments and message clarity in scripts and test files.
  
- **Refactor**
  - Updated error messages and variable names to enhance code readability and maintainability.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,9 @@ Please follow the recommendations outlined at [keepachangelog.com](http://keepac
 ### [Unreleased]
```
[the remark this answers is not in the document]
Comment (G-Rath) [reply CHANGELOG.md:18]: out of scope of this PR

File: CHANGELOG.md [hunk 2]
```diff
@@ -174,7 +177,7 @@
 
 #### Changed
```
[the remark this answers is not in the document]
Comment (G-Rath) [reply CHANGELOG.md:178]: out of scope of this PR

File: CHANGELOG.md [hunk 3]
```diff
@@ -201,7 +204,7 @@
 - Added the ability to assign a module with a `call` method to `config.build_production_command`. See [the configuration docs](https://www.shakacode.com/react-on-rails/docs/guides/configuration). [PR 1362: Accept custom module for config.build_production_command](https://github.com/shakacode/react_on_rails/pull/1362).
 
 #### Fixed
```
[the remark this answers is not in the document]
Comment (G-Rath) [reply CHANGELOG.md:206]: out of scope of this PR

File: CHANGELOG.md [hunk 4]
```diff
@@ -756,7 +759,7 @@
 - Don't assume ActionMailer is available. [#608](https://github.com/shakacode/react_on_rails/pull/608) by [tuzz](https://github.com/tuzz).
 
 ## [6.2.1] - 2016-11-19
```
[the remark this answers is not in the document]
Comment (G-Rath) [reply CHANGELOG.md:761]: out of scope of this PR

File: CHANGELOG.md [hunk 5]
```diff
@@ -782,12 +785,12 @@
 
 ##### Fixed
```
[the remark this answers is not in the document]
Comment (G-Rath) [reply CHANGELOG.md:786]: out of scope of this PR

File: CHANGELOG.md [hunk 6]
```diff
@@ -380,7 +383,7 @@
 
 ### [11.0.7] - 2018-05-16
```
[the remark this answers is not in the document]
Comment (G-Rath) [reply CHANGELOG.md:384]: out of scope of this PR

File: CHANGELOG.md [hunk 7]
```diff
@@ -903,7 +906,7 @@
 ## [4.0.3] - 2016-03-17
 
 ##### Fixed
```
[the remark this answers is not in the document]
Comment (G-Rath) [reply CHANGELOG.md:908]: out of scope of this PR

File: spec/dummy/client/README.md [hunk 8]
```diff
@@ -18,7 +18,7 @@ or to autofix
     yarn run lint -- --fix
```
[the remark this answers is not in the document]
Comment (G-Rath) [reply spec/dummy/client/README.md:20]: out of scope of this PR

File: spec/react_on_rails/utils_spec.rb [hunk 9]
```diff
@@ -294,7 +294,7 @@ module ReactOnRails
               allow(Rails).to receive(:version).and_return("4.2")
             end
 
-            it "memoizes the result" do
+            it "memorizes the result" do
```
Comment (justin808) [inline spec/react_on_rails/utils_spec.rb]: memoize was correct.
```suggestion
            it "memoizes the result" do
```

**Review by justin808 — APPROVED** [review]

**Comment by justin808** [comment]
Thanks so much @G-Rath!

**Pull request merged**